command/agent_test.go

```diff
@@ -301,6 +305,7 @@ func TestAgent_HostBasedIDs(t *testing.T) {
 		cmd := &AgentCommand{
 			args: []string{
 				"-data-dir", tmpDir,
+				"-bind", "127.0.0.1",
```

Comment:
this calls readConfig which previously didn't fully initialize the config ( so it left out validation errors caused by calling ResolveTmplAddrs and SetupTaggedAndAdvertiseAddrs). Previously, it would end up failing downstream when a new agent is constructed. Now that I've moved that code to readConfig, it needs the bind address for readConfig to succeed. 
This test now also matches the minimal set of args needed to get consul agent running.